### PR TITLE
[WIP] Use PageFind for search page

### PR DESCRIPTION
**Pull request by sftim**
This (draft) PR adapts https://github.com/kubernetes/website/pull/44530 so that we use it for the existing search function.

[preview](https://deploy-preview-44655--kubernetes-io-main-staging.netlify.app/search/?q=Kubernetes)

/area web-development

### CODE REVIEW

**Comment by sftim** [comment]
Relevant to issue https://github.com/kubernetes/website/issues/44475

File: layouts/case-studies/single-baseof.html [hunk 1]
```diff
@@ -6,7 +6,9 @@
 	<body>
 		{{ partial "navbar.html" . }}
 		{{ partial "deprecation-warning.html" . }}
+                <div data-pagefind-body>
 		{{ block "main" . }}{{ end }}
+                <div>
```
Comment (Gauravpadam) [inline layouts/case-studies/single-baseof.html:11]:
```suggestion
                </div>
```

**Review by Gauravpadam — COMMENTED** [review]
The overview looks promising, I'll test this out

**Comment by nate-double-u** [comment]
/remove-lifecycle stale

This is still a relevant exploration.

**Comment by sftim** [comment]
Anything you want to collaborate on here @nate-double-u ? I can make some time to run through my thinking.

**Comment by nate-double-u** [comment]
I would very much appreciate that @sftim, thanks!

I'll switch over to Kubernetes slack to set something up.

**Comment by sftim** [comment]
/close

No longer needed

**Pull request closed without merging**